runners/google-cloud-dataflow-java/build.gradle

```diff
@@ -0,0 +1,95 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply from: project(":").file("build_rules.gradle")
+applyJavaNature(enableFindbugs: false /* BEAM-925 */)
+
+description = "Apache Beam :: Runners :: Google Cloud Dataflow"
+
+/*
+ * We need to rely on manually specifying these evaluationDependsOn to ensure that
+ * the following projects are evaluated before we evaluate this project. This is because
+ * we are attempting to reference the "sourceSets.test.output" directly.
+ * TODO: Swap to generating test artifacts which we can then rely on instead of
+ * the test outputs directly.
+ */
+evaluationDependsOn(":beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-io-parent:beam-sdks-java-io-google-cloud-platform")
+evaluationDependsOn(":beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-extensions-parent:beam-sdks-java-extensions-google-cloud-platform-core")
+evaluationDependsOn(":beam-model-parent:beam-model-fn-execution")
+
+processResources {
+  filter org.apache.tools.ant.filters.ReplaceTokens, tokens: [
+    'dataflow.legacy_environment_major_version' : '6',
+    'dataflow.fnapi_environment_major_version' : '1',
+    'dataflow.container_version' : 'beam-master-20170926'
+  ]
+}
+
+// Exclude tests that need a runner
+test {
+  systemProperty "beamTestPipelineOptions", ""
+  systemProperty "beamUseDummyRunner", "true"
+}
+
+dependencies {
+  compile library.java.guava
+  shadow project(path: ":beam-model-parent:beam-model-pipeline", configuration: "shadow")
+  shadow project(path: ":beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-core", configuration: "shadow")
+  shadow project(path: ":beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-extensions-parent:beam-sdks-java-extensions-google-cloud-platform-core", configuration: "shadow")
+  shadow project(path: ":beam-sdks-parent:beam-sdks-java-parent:beam-sdks-java-io-parent:beam-sdks-java-io-google-cloud-platform", configuration: "shadow")
+  shadow project(path: ":beam-runners-parent:beam-runners-core-construction-java", configuration: "shadow")
+  shadow library.java.google_api_client
+  shadow library.java.google_http_client
+  shadow library.java.google_http_client_jackson2
+  shadow library.java.google_api_services_dataflow
+  shadow library.java.google_api_services_clouddebugger
+  shadow library.java.google_api_services_storage
+  shadow library.java.google_auth_library_credentials
+  shadow library.java.google_auth_library_oauth2_http
+  shadow library.java.bigdataoss_util
+  shadow library.java.avro
+  shadow library.java.joda_time
```

Comment:
<!--new_thread; commit:343c7ce935700e9f2a23ca261db2af2c05640b27; resolved:0-->
Is this shading way more than we do today? Or is this not shaded and shadow means something else?

Reply:
Its the opposite of what you would expect since the shadow plugin conveniently by default decides to bundle everything instead of bundling nothing. Marking something with `shadow` or the equivalent test one means you want to keep it. Anything in the runtime/testRuntime configurations is meant to be shaded and removed from the final artifact.